pandas/io/pytables.py

```diff
@@ -2094,13 +2097,22 @@ def convert(self, values, nan_rep, encoding):
     def get_attr(self):
         """ get the data for this colummn """
         self.values = getattr(self.attrs, self.kind_attr, None)
+        if self.values is None:
```

Comment:
you don't need to actually set ``_handle``, you can get it directly from ``self.table`` (maybe make a property to do this), I think its slightly cleaner